### PR TITLE
Allow type=number and password on Input component

### PR DESCRIPTION
**Pull request by acelaya**

### CODE REVIEW

File: src/components/input/Input.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { downcastRef } from '../../util/typing';
 import InputRoot from './InputRoot';
 
 type ComponentProps = {
-  type?: 'email' | 'search' | 'text' | 'url';
+  type?: 'email' | 'search' | 'text' | 'url' | 'number';
```
Comment (robertknight) [inline src/components/input/Input.tsx:8]: I suggest to sort these values to make the list easier to scan.

**Review by robertknight — APPROVED** [review]

File: src/components/input/Input.tsx (hunk 1 above)
Comment (robertknight) [reply src/components/input/Input.tsx:8]: Also, are there other values for `type` that we should be enabling? See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types for a list.

Comment (acelaya) [reply src/components/input/Input.tsx:8]: > I suggest to sort these values to make the list easier to scan.

Sort how?

Comment (acelaya) [reply src/components/input/Input.tsx:8]: > Also, are there other values for type that we should be enabling? See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types for a list.

Perhaps it's worth having `password` as well.

Comment (acelaya) [reply src/components/input/Input.tsx]: > > I suggest to sort these values to make the list easier to scan.
> 
> Sort how?

I have rearranged them a bit, putting text first, as it's the most common and default value.

**Pull request merged**